### PR TITLE
BHoM_Engine: Created DocumentationURL

### LINKED ISSUE #3027: Modify all website urls within BHoM related repositories to reference a central location
#### Description:

Links to websites as exposed through buttons in each UI should be located in a central location so that a links can be updated across all UIs in one modification.

### PR DESCRIPTION
**Pull request by desaiwangBH**
### Issues addressed by this PR

Closes #3027 

Added a method to create links to the BHoM wiki documentation.

### Changelog

- Added a method to create links to the BHoM wiki documentation.

### CODE REVIEW

File: BHoM_Engine/Query/GetWikiURL.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+{
+    public static partial class Query
+    {
+
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("Generates wiki url given provided page.")]
+        [Input("page", "The page for the url.")]
```
Comment (FraserGreenroyd) [inline BHoM_Engine/Query/GetWikiURL.cs:36]:
```suggestion
        [Input("page", "An optional page on the documentation to link to. If no page is provided, the root documentation URL is returned.")]
```

File: BHoM_Engine/Query/GetWikiURL.cs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+namespace BH.Engine.Base
+{
+    public static partial class Query
+    {
+
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("Generates wiki url given provided page.")]
```
Comment (FraserGreenroyd) [inline BHoM_Engine/Query/GetWikiURL.cs:35]:
```suggestion
        [Description("Generates the wiki URL for a provided page.")]
```

File: BHoM_Engine/Query/GetWikiURL.cs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        [Input("page", "The page for the url.")]
+        [Output("url", "Fully qualified url for BHoM wiki documentation.")]
```
Comment (FraserGreenroyd) [inline BHoM_Engine/Query/GetWikiURL.cs:37]:
```suggestion
        [Output("url", "Fully qualified URL for BHoM wiki documentation.")]
```

File: BHoM_Engine/Query/GetWikiURL.cs [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+            string url = "https://bhom.xyz/documentation";
+
+            if (!string.IsNullOrEmpty(page))
+            {
+                url += $"/{page}";
+            }
+
+            return url;
+        }
+
```
Comment (FraserGreenroyd) [inline BHoM_Engine/Query/GetWikiURL.cs:49]:
```suggestion
```

**Review by FraserGreenroyd — CHANGES_REQUESTED** [review]
I was thinking about this - we don't call the documentation a wiki, we refer to it as documentation, and I wonder if therefore the method should reflect that as well? One to discuss in addition to the comments on the code, but great start @desaiwangBH 😄

**Comment by desaiwangBH** [comment]
@BHoMBot check compliance

**Comment by desaiwangBH** [comment]
@BHoMBot check required

**Review by FraserGreenroyd — APPROVED** [review]
This looks good to me now, thanks @desaiwangBH

**Comment by FraserGreenroyd** [comment]
@BHoMBot check ready-to-merge

**Comment by FraserGreenroyd** [comment]
@BHoMBot this is a DevOps instruction. I am requesting neutral checks on: unit-tests

**Comment by FraserGreenroyd** [comment]
@BHoMBot check core
@BHoMBot check compliance

**Comment by FraserGreenroyd** [comment]
@BHoMBot this is a DevOps instruction. I am requesting neutral checks on: versioning, installer, serialisation, ready-to-merge, null-handling, unit-tests

**Pull request merged**